core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -225,7 +227,7 @@ public ProcessingType getProcessingType() {
       };
     } else {
       return messagingException -> Mono.from(getMessagingExceptionHandler().apply(messagingException))
-          .flatMap(handled -> {
+          .flatMapMany(handled -> {
```

Comment:
@dfeist please check this change. This should also be changed in other places as the method was deprecated.